fuel/src/test/kotlin/com/github/kittinunf/fuel/RequestUploadTest.kt

```diff
@@ -235,4 +231,45 @@ class RequestUploadTest : BaseTestCase() {
         val statusCode = HttpURLConnection.HTTP_OK
         assertThat(response?.httpStatusCode, isEqualTo(statusCode))
     }
+
+    @Test
+    fun httpUploadWithDataParts() {
+        var request: Request? = null
+        var response: Response? = null
+        var data: Any? = null
+        var error: FuelError? = null
+
+        manager.upload("/post", param = listOf("foo" to "bar"))
+                .sources { request, url ->
+                    listOf(File(currentDir, "lorem_ipsum_short.tmp"),
+                            File(currentDir, "lorem_ipsum_long.tmp"))
+                }
+                // Override with Data parts
+                .dataParts { request, url ->
+                    listOf(DataPart("first-file", "image/jpeg", File(currentDir, "lorem_ipsum_short.tmp")),
```

Comment:
And I suggest that we have one overloading method. in the case, where user just want to upload one file, so it should be extremely simple.

Like
``` Kotlin
Fuel.upload("/upload").source { req, url -> File(currentDir, "foo.txt") }.responseString()
```

Maybe we can construct DataPart object from a File object. For example, if the File is supplied, we can infer `name` by using the filename.